components/sidebar/sidebar_channel/sidebar_channel_menu/sidebar_channel_menu.tsx

```diff
@@ -31,133 +31,191 @@ type Props = PropsFromRedux & OwnProps;
 
 const SidebarChannelMenu = (props: Props) => {
     const isLeaving = useRef(false);
-    const [openUp, setOpenUp] = useState(false);
 
     const {formatMessage} = useIntl();
 
-    let markAsReadMenuItem: JSX.Element | null = null;
+    let markAsReadUnreadMenuItem: JSX.Element | null = null;
     if (props.isUnread) {
-        function handleMarkAsRead() {
+        function handleMarkAsRead(event: MouseEvent<HTMLLIElement>) {
+            event.preventDefault();
+
             props.markChannelAsRead(props.channel.id);
             trackEvent('ui', 'ui_sidebar_channel_menu_markAsRead');
         }
 
-        markAsReadMenuItem = (
-            <Menu.ItemAction
+        markAsReadUnreadMenuItem = (
+            <Menu.Item
                 id={`markAsRead-${props.channel.id}`}
                 onClick={handleMarkAsRead}
-                icon={<MarkAsUnreadIcon size={16}/>}
-                text={formatMessage({id: 'sidebar_left.sidebar_channel_menu.markAsRead', defaultMessage: 'Mark as Read'})}
+                leadingElement={<MarkAsUnreadIcon size={18}/>}
+                labels={
+                    <FormattedMessage
+                        id='sidebar_left.sidebar_channel_menu.markAsRead'
+                        defaultMessage='Mark as Read'
+                    />
+                }
             />
+
         );
-    }
+    } else {
+        function handleMarkAsUnread(event: MouseEvent<HTMLLIElement>) {
+            event.preventDefault();
 
-    let markAsUnreadMenuItem: JSX.Element | null = null;
-    if (!props.isUnread) {
-        function handleMarkAsUnread() {
             props.markMostRecentPostInChannelAsUnread(props.channel.id);
             trackEvent('ui', 'ui_sidebar_channel_menu_markAsUnread');
         }
 
-        markAsUnreadMenuItem = (
-            <Menu.ItemAction
+        markAsReadUnreadMenuItem = (
+            <Menu.Item
                 id={`markAsUnread-${props.channel.id}`}
                 onClick={handleMarkAsUnread}
-                icon={<MarkAsUnreadIcon size={16}/>}
-                text={formatMessage({id: 'sidebar_left.sidebar_channel_menu.markAsUnread', defaultMessage: 'Mark as Unread'})}
+                leadingElement={<MarkAsUnreadIcon size={18}/>}
+                labels={
+                    <FormattedMessage
+                        id='sidebar_left.sidebar_channel_menu.markAsUnread'
+                        defaultMessage='Mark as Unread'
+                    />}
             />
         );
     }
 
-    let favoriteMenuItem: JSX.Element | null = null;
+    let favoriteUnfavoriteMenuItem: JSX.Element | null = null;
```

Comment:
naming is a bit off here, its a toggle, soooooo

```suggestion
    let favoriteMenuItemToggle: JSX.Element | null = null;
```

Reply:
Hey, Thank you for the feedback, but i would like to leave this as is. Hope you don't mind.